### PR TITLE
#5801 Differentiate GoogleAuthErr And EnterpriseServerAuthErr

### LINKED ISSUE #5801: we should differentiate `GoogleAuthErr` and `EnterpriseServerAuthErr`. When custom IdP is used and it is `EnterpriseServerAuthErr`, we should be showing `ConfiguredIdpOauth` popup instead of `GoogleOAuth` popup

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR implemented function to refresh custom idp id token if expires and show custom idp auth popup when auth error occurs

close #5801 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Test added

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/api/shared/api-error.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ interface RawAjaxErr {
 
 abstract class AuthErr extends Error {}
 export class GoogleAuthErr extends AuthErr {}
+export class EnterpriseServerAuthErr extends AuthErr {}
 export class BackendAuthErr extends AuthErr {}
```
Comment (sosnovsky) [inline extension/js/common/api/shared/api-error.ts:31]: Are there any differences between `EnterpriseServerAuthErr` and `BackendAuthErr`? Can we merge them into single `EnterpriseServerAuthErr`?

**Comment by ioanmo226** [comment]
Ohh. Yep
It's necessary to add UI test.
Let me add that.

File: extension/js/common/api/shared/api-error.ts (hunk 1 above)
Comment (ioanmo226) [reply extension/js/common/api/shared/api-error.ts:31]: Yeah. I think we can

**Review by sosnovsky — APPROVED** [review]
Well done, works great 👍

**Pull request merged**